web3/_utils/filters.py

```diff
@@ -78,9 +79,9 @@ def construct_event_filter_params(
     address: Optional[ChecksumAddress] = None,
 ) -> Tuple[List[List[Optional[HexStr]]], FilterParams]:
     filter_params: FilterParams = {}
-    topic_set: Sequence[HexStr] = construct_event_topic_set(
-        event_abi, abi_codec, argument_filters
-    )
+    topic_set: Union[
```

Comment:
It's not clear to me where the nested list would come from. Is that potentially from the `topics` argument? Or should the return type on `construct_event_topic_set` be updated?

In either case I think the type would also be `Union[Sequence[HexStr], Sequence[Sequence[HexStr]]]`.

Reply:
I did not dig into `construct_event_topic_set` output. I just took the check on line 94 to mean it sometimes comes back as a nested list. But I suppose yes, if the return type of `List[HexStr]` is to be taken seriously, I should match it here.

Reply:
Decided to take `construct_event_topic_set` at its word and leave `topic_set` as `Sequence[HexStr`. If there are cases where it comes back as a nested list, we should address that as a bug and not just have a :shrug:  check here.